File: auto-docs/executables/python/axes_range_mode.py
import plotly.plotly as py
from plotly.graph_objs import *

py.sign_in('TestBot', 'r1neazxo9w')

data = Data([
    Scatter(
        x=[2, 4, 6],
        y=[-3, 0, 3]
    )
])
layout = Layout(
    showlegend=False,
    xaxis=XAxis(
        rangemode='tozero',
        autorange=True
    ),
    yaxis=YAxis(
        rangemode='nonnegative',
        autorange=True
    )
)
fig = Figure(data=data, layout=layout)

plot_url = py.plot(fig, filename='axes-range-mode', auto_open=False)

File: auto-docs/executables/node/axes_range_mode.js
var plotly = require('plotly')('theengineear', 'o9zlr0hy6z')

var data = [
  {
    x: [2, 4, 6], 
    y: [-3, 0, 3], 
    type: "scatter"
  }
];
var layout = {
  xaxis: {
    autorange: true, 
    rangemode: "tozero"
  }, 
  yaxis: {
    autorange: true, 
    rangemode: "nonnegative"
  }, 
  showlegend: false
};

var graph_options = {filename: "axes-range-mode", fileopt: "overwrite", layout: layout, auto_open: "false"}
plotly.plot(data, graph_options, function (err, msg) {
    console.log(msg);
});